### PR TITLE
Adds support for Private Messaging for users on a pod

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

This basically implements a form of "Direct Messaging" or "Priaving
Messaging", we'll call it the later.

Right now this only implements support for "on-pod" messaging. Cross-pod
will come later when we work out how to "mutually authenticate" between
pods.

The underlying mechanisms are:

- SMTP for inter-pod delivery and user access (_only the later of which
  is tested_).
- POP3 for message access by authenticated users.
- Mailbox file format for storage

And of course there is a Web Interface that wraps all this in a nice
way. An API will come later when we start to think about adding this to
the Mobile App (Goryon).

##### Component Name

- area/backend
- area/ui

##### Test Plan

Tested locally extensively as it was develoepd.

Mostly independent code so existing features are not affected in any
major way and should continue to work as expected.

##### Additional Information

### CODE REVIEW

File: internal/messages_handler.go.orig [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+// ListMessagesHandler ...
+func (s *Server) ListMessagesHandler() httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		ctx := NewContext(s.config, s.db, r)
+
+		ctx.Title = "Private Messages"
+
+		ctx.Messages = Messages{
+			&Message{
+				From: "kate",
```
Comment (aaadonai) [inline internal/messages_handler.go.orig:21]: Is this testing data?

File: internal/messages_handler.go.orig [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	}
+}
+
+// ViewMessageHandler ...
+func (s *Server) ViewMessageHandler() httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		ctx := NewContext(s.config, s.db, r)
+
+		if r.Method == http.MethodPost {
+			// TODO: Implement this :D
```
Comment (aaadonai) [inline internal/messages_handler.go.orig:96]: Is this missing or will come later?

File: internal/messages_handler.go.orig [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			return
+		}
+
+		ctx.Title = "Private Message from Kate: Hello"
```
Comment (aaadonai) [inline internal/messages_handler.go.orig:100]: this seems like testing data to me? Is it?

File: internal/messages_handler.go.rej [hunk 4]
```diff
@@ -0,0 +1,101 @@
+***************
```
Comment (aaadonai) [inline internal/messages_handler.go.rej:1]: ok, looks like .rej file and .orig file were committed by mistake? Am I right?

File: internal/options.go.orig [hunk 5]
```diff
@@ -0,0 +1,439 @@
+package internal
```
Comment (aaadonai) [inline internal/options.go.orig:1]: another .orig file.

File: internal/options.go.rej [hunk 6]
```diff
@@ -0,0 +1,34 @@
+***************
```
Comment (aaadonai) [inline internal/options.go.rej:1]: another .rej file

**Review by aaadonai — APPROVED** [review]
I did my best to review this to the best of my abilities! ;-)

File: internal/messages_handler.go.orig (hunk 1 above)
Comment (prologic) [reply internal/messages_handler.go.orig:21]: Oh good catch!

File: internal/messages_handler.go.orig (hunk 2 above)
Comment (prologic) [reply internal/messages_handler.go.orig:96]: Nice catch! This got moved to another handler

File: internal/messages_handler.go.orig (hunk 3 above)
Comment (prologic) [reply internal/messages_handler.go.orig:100]: Yeah I'll check later today

File: internal/messages_handler.go.rej (hunk 4 above)
Comment (prologic) [reply internal/messages_handler.go.rej:1]: Looks like it!

**Comment by prologic** [comment]
Thanks!

**Comment by prologic** [comment]
@aaadonai Those were all accidentally committed files from bad merge conflict resolution and didn't notice them left around. Fixed and added to `.gitignore` 👍

**Pull request merged**